tests/resources/deriv_approx_sin.xml

```diff
@@ -3,7 +3,7 @@
 
   <component name="sin" id="sin">
     <variable id="x" units="dimensionless" name="x" interface="public_and_private"/>
-    <variable id="sin" units="dimensionless" name="sin" initial_value="sin_initial_value" interface="public_and_private"/>
+    <variable id="sin" units="dimensionless" name="sin" initial_value="0.0" interface="public_and_private"/>
```

Comment:
This is now testing something different ... even if we are phasing out the initialisation with variable thing, we should keep the tests?

Reply:
This was @nickerso's change and I think it's fine since we are not currently handling the initialisation of a variable from another variable.

Reply:
Yes we are ... that's still permitted, just to be noted in the tutorials and informative spec that it isn't recommended and will be removed soon.  If it's changed here, we need to check that we still have tests for initialisation from a variable until it's actually removed properly ...

Reply:
I only meant that the code generator doesn't currently support the initialisation of a variable using another variable. This is clearly something that we need to add to the code generator. I have just created an issue for it (see issue #602).

Reply:
Ah, ok, that makes more sense.  I thought we were removing functionality rather than just waiting for it :)